runtime/runtime/src/lib.rs

```diff
@@ -1853,6 +1883,82 @@ impl TotalResourceGuard {
     }
 }
 
+struct PrefetchManager {
+    prefetcher: Option<TriePrefetcher>,
+    look_ahead_len: usize,
+    next_delayed_receipt_index: usize,
+    next_local_receipt_index: usize,
+    next_incoming_receipt_index: usize,
+}
+
+// TODO remove `look_ahead_len` here, instead just prefetch one next receipt. less edge cases.
+impl PrefetchManager {
```

Comment:
can this data and logic be just part of the TriePrefetcher instead of having a separate manager on top of it? I see the TriePrefetcher is already aware of receipts and transactions and it can be aware of types of receipts as well (local, delayed, incoming)?

Reply:
PrefetchManager is scoped to an invocation of `fn apply`, while TriePrefetcher is scoped to [the transaction runtime](https://github.com/mooori/nearcore/blob/2e89a490d2a7e963752c4dd5d1b3d8df7bdfa0c9/runtime/runtime/src/prefetch.rs#L58), which I think is a broader scope. My interpretation is that integrating PrefetchManager functionality into TriePrefetcher would require limiting TriePrefetcher's scope to a particular `fn apply` invocation as well. 

The current PrefetchManager turned out to be a suboptimal abstraction and I'm working on some changes of how delayed receipts are handled. If this results in a performance improvement and the PR is not discarded, I'll consider moving things into TriePrefetcher.